### PR TITLE
[volk-]gnss-sdr: Remove -next subports

### PR DESCRIPTION
**Pull request by carlesfernandez**
#### Description

As discussed in #5915, remove -next subports. Only  [volk-]gnss-sdr and [volk-]gnss-sdr-devel remain.

This change alleviates CI workload when checking changes in the gnss-sdr Portfile.

The gnss-sdr-devel  and volk-gnss-sdr-devel subports are now built from the [next](https://github.com/gnss-sdr/gnss-sdr/tree/next) branch of the gnss-sdr upstream repo, where development takes place. This new nomenclature is better aligned with existing practice in Macports.

Now gnss-sdr and gnss-sdr-devel subports can be built against any gnuradio subport available.

@michaelld : 

Please note that I’m not able to build the gnuradio-devel subport  in my machine due to the following error:

```
info:build /opt/local/var/macports/build/_opt_local_var_macports_sources_rsync.macports.org_release_tarballs_ports_science_gnuradio/gnuradio-devel/work/gnuradio-896d1c9da31963ecf5b0d90942c2af51ca998a69/gr-blocks/lib/socket_pdu_impl.cc:168:82: error: no member named 'get_io_service' in 'boost::asio::basic_socket_acceptor<boost::asio::ip::tcp, boost::asio::executor>'
:info:build       tcp_connection::sptr new_connection = tcp_connection::make(d_acceptor_tcp->get_io_service(), d_rxbuf.size(), d_tcp_no_delay);
:info:build                                                                  ~~~~~~~~~~~~~~  ^
```

Tests against gnuradio and gnuradio-next work fine.

###### Type(s)

- [ ] bugfix
- [x] enhancement
- [ ] security fix

###### Tested on

macOS 10.15.1
Xcode 11.2.1

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [ ] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL?

- [x] checked your Portfile with `port lint`?
- [x] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?

### CODE REVIEW

**Comment by michaelld** [comment]
@carlesfernandez ACK about the `gnuradio-devel` build issue ... there's an issue open for it ... I need to fix it.

**Comment by carlesfernandez** [comment]
Force-pushed a commit with -next subports `replaced_by` -devel.

**Comment by carlesfernandez** [comment]
mmm according to the CI, I didn't `replaced_by` correctly. How it should be done?

**Comment by carlesfernandez** [comment]
I'm trying with

```
# This port can be removed on Dec 4, 2020.
subport gnss-sdr-next {
    replaced_by gnss-sdr-devel
    PortGroup obsolete 1.0

    name      gnss-sdr-next
    version   20191204
    revision  0
}
```

but I still get an error message:

```
$ sudo port install gnss-sdr-next
--->  Configuring gnss-sdr-next
Error: gnss-sdr-next has been replaced by gnss-sdr-devel; please install that instead.
Error: Failed to configure gnss-sdr-next: obsolete port
Error: See /opt/local/var/macports/logs/_opt_local_var_macports_sources_rsync.macports.org_release_tarballs_ports_science_gnss-sdr/gnss-sdr-next/main.log for details.
Error: Follow https://guide.macports.org/#project.tickets to report a bug.
Error: Processing of port gnss-sdr-next failed
```

Is this expected behavior?

**Comment by michaelld** [comment]
Yes that's the expected behavior. I find it a bit awkward, but it makes sense: the user needs to know what's going on and take action, decide to move to the new port, rather than us just forcing it on them. Or something like that.

**Comment by carlesfernandez** [comment]
You are right: `gnss-sdr` cannot use `volk-gnss-sdr-devel`, and `gnss-sdr-devel` cannot use `volk-gnss-sdr`.

I've just force-pushed a commit removing redundant conflicts.

**Comment by carlesfernandez** [comment]
Another force-push with a patch fixing gtest building if the compiler uses C++98 standard by default.

File: science/gnss-sdr/Portfile [hunk 1]
```diff
@@ -220,20 +223,23 @@ if {![variant_isset opencl]} {
 
 variant osmosdr description "Install ${name} with support for OsmoSDR and other front-ends (RTL-based dongles, HackRF, bladeRF, etc.) as signal source." {
 
-    if {${subport} ne "gnss-sdr-next"} {
+    if {[catch {set installed [lindex [registry_active gnuradio-next] 0]}]} {
```
Comment (michaelld) [inline science/gnss-sdr/Portfile]: After testing, this "if" isn't necessary any longer. Please just leave the internal code.

Comment (carlesfernandez) [reply science/gnss-sdr/Portfile]: Done

**Pull request merged**